DHCP6 init before RA option

marjohn56 opened the pull request:
Some ISP BNG's require tha a dhcp6 init is sent befire RA. The additions in these two files handle that option. The interfaces.php has changes that create an extra option in the DHCP6 WAN config - adding an the option 'Do not wait for a RA'; this will set or unset the flag 'dhcp6withoutra'.

interfaces.inc uses the flag set in the configuration to decide whether to launch dhcp6c using the rtsoldscript or to launch it without waiting for an RA that would trigger the script.

Other additions to the interfaces.inc attempt to consolidate the multiple calls and methods of killing the dhcp6c client into a single call, much the same as is used with the dhcp4 client. The only place this cannot be done is where rtsold launches dhcp6c and the script itself must kill the client, if it is running.

cbuechler commented on src/etc/inc/interfaces.inc:
```diff
@@ -4004,9 +4016,25 @@ function interface_dhcpv6_configure($interface = "wan", $wancfg) {
 	/* Add wide-dhcp6c shell script here. Because we can not pass a argument to it. */
 	if (!@file_put_contents("{$g['varetc_path']}/rtsold_{$wanif}_script.sh", $rtsoldscript)) {
-		printf("Error: cannot open rtsold_{$wanif}_script.sh in interface_dhcpv6_configure() for writing.\n");
+		printf("Error: cannot open rtsold_{$interface}_script.sh in interface_dhcpv6_configure() for writing.\n");
```
this is overwriting a bug fix, needs to be $wanif

cbuechler commented on src/etc/inc/interfaces.inc:
```diff
@@ -4004,9 +4016,25 @@ function interface_dhcpv6_configure($interface = "wan", $wancfg) {
+		$rtsoldscript .= "fi\n";
+	}
+	else
+	{
```
Needs to be "} else {" on one line to match our style guide.

cbuechler commented on src/etc/inc/interfaces.inc:
```diff
@@ -4022,12 +4050,20 @@ function interface_dhcpv6_configure($interface = "wan", $wancfg) {
 	}
+	if (isset($wancfg['dhcp6withoutra'])){
+		kill_dhcp6client_process($wanif);	
+		sleep(3);
```
Is there any reason to sleep here? Seems that just adds unnecessary delay. There is already a sleep in kill_dhcp6client_process to give it time to die off, so it seems like it should be safe to move on immediately here without the sleep.

cbuechler commented on src/etc/inc/interfaces.inc:
```diff
@@ -4004,9 +4016,25 @@ function interface_dhcpv6_configure($interface = "wan", $wancfg) {
 	$rtsoldscript .= "/usr/local/sbin/dhcp6c {$debugOption} -c {$g['varetc_path']}/dhcp6c_{$interface}.conf -p {$g['varrun_path']}/dhcp6c_{$wanif}.pid {$wanif}\n";
 	$rtsoldscript .= "/usr/bin/logger -t rtsold \"Starting dhcp6 client for interface {$interface}({$wanif})\"\n";
+	/* non ipoe Process */
+ 	if (!isset($wancfg['dhcp6withoutra'])){	
```
style, should be ") {" at the end of the line. space before the {

cbuechler commented:
Thanks for the update @marjohn56. Left some comments in the commit. 

You still ended up with an unnecessary merge commit, seemingly didn't rebase origin. Maybe that's how you ended up reverting the one bug fix I noted.

phil-davis commented on src/etc/inc/interfaces.inc:
```diff
@@ -4004,9 +4016,25 @@ function interface_dhcpv6_configure($interface = "wan", $wancfg) {
+		$rtsoldscript .= "\t/bin/sleep 1\n";
+	}
+	$debugOption = isset($wancfg['dhcp6debug']) ? "-D" : "-d";
+	if (!isset($wancfg['dhcp6withoutra'])){
```
More little style stuff
s/){/) {
There are a couple of places that need the space between ")" and "{".

marjohn56 commented:
Hi Chris,

 

I think it was more a case of haste; the changes I have made had been running on my device for a week or more for testing.

 

I seem to read one thing with regards to GIT and then read something that conflicts with that, hence the commit.

 

Here’s how I did this.

 

Create a branch from pfSense. 

 

In that branch I then created a new branch DHCP6 to work on. I then copied my working files from the live device and THIS is where the overwrite of the bug fix for $interface occurred. I need to be more careful in comparing the branches 

 

I then committed the changes I had made back to the first branch and then issued a pull request.

 

I note the comment on the extra sleep and that can come out, I had forgotten that I had put a sleep call in the kill process.

 

I will make the changes and the style corrections an re-submit.

 

Regards,

 

Martin

 

 

 

 

 

From: Chris Buechler [mailto:notifications@github.com] 
Sent: 10 July 2016 03:40
To: pfsense/pfsense <pfsense@noreply.github.com>
Cc: marjohn56 <martin@queens-park.com>; Mention <mention@noreply.github.com>
Subject: Re: [pfsense/pfsense] DHCP6 init before RA option (#3053)

 

Thanks for the update @marjohn56 <https://github.com/marjohn56> . Left some comments in the commit. 

You still ended up with an unnecessary merge commit, seemingly didn't rebase origin. Maybe that's how you ended up reverting the one bug fix I noted. 

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub <https://github.com/pfsense/pfsense/pull/3053#issuecomment-231567362> , or mute the thread <https://github.com/notifications/unsubscribe/ARIXpoAGosd2PKSt2A3WJH1J3igQIEpaks5qUFucgaJpZM4JImZv> .  <https://github.com/notifications/beacon/ARIXpsQEPud6b8Z_Ruok62zLpndoxJ7Jks5qUFucgaJpZM4JImZv.gif>

marjohn56 commented:
I have updated the interfaces.inc correcting the style using the online editor here, I have corrected the overwrite, removed the superfluous sleep() call I have also corrected the style.

Do I need to issue a new pull request.

phil-davis commented:
It is much cleaner for the future if "master" just gets a single commit. Since there are only 2 files changed, it will be easy if you make a new branch from master, put the 2 files in it, make a single commit and new PR from that.

marjohn56 commented:
Thanks Phil,  I will do what you suggest in future. Although I can find my way around various programming languages, GIT is an environment I am still coming to terms with.

phil-davis commented:
Yes, then Chris can do final review and commit easily.

marjohn56 commented:
OK, well I think I have followed your instructions….

 

Guess I’ll find out soon enough!

 

Regards,

 

Martin

 

From: Phil Davis [mailto:notifications@github.com] 
Sent: 10 July 2016 15:47
To: pfsense/pfsense <pfsense@noreply.github.com>
Cc: marjohn56 <martin@queens-park.com>; Mention <mention@noreply.github.com>
Subject: Re: [pfsense/pfsense] DHCP6 init before RA option (#3053)

 

Yes, then Chris can do final review and commit easily.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub <https://github.com/pfsense/pfsense/pull/3053#issuecomment-231592686> , or mute the thread <https://github.com/notifications/unsubscribe/ARIXpgozKBzJVWBi92fNY72BdgkKCR88ks5qUQXngaJpZM4JImZv> .

phil-davis commented:
Looks good in #3055 - I made a couple of comments there on minor code style.
